### PR TITLE
Add `unplayed=` song select filter query

### LINKED ISSUE #29568: Unplayed= does not work while searcing for beatmaps in osu! laser
### Type

Game behaviour

### Bug description

When trying to find unplayed beatmaps i normally use unplayed= in stable osu, but currently in osu! laser this feature does not work.

### Screenshots or videos

image

### Version

2024.817.0

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/16715563/compressed-logs.zip)

### PR DESCRIPTION
**Pull request by smoogipoo**
Resolves https://github.com/ppy/osu/issues/29568

### CODE REVIEW

File: osu.Game/Screens/Select/FilterQueryParser.cs [hunk 1]
```diff
@@ -66,6 +66,10 @@ private static bool tryParseKeywordCriteria(FilterCriteria criteria, string key,
                 case "lastplayed":
                     return tryUpdateDateAgoRange(ref criteria.LastPlayed, op, value);
 
+                case "unplayed":
+                    criteria.LastPlayed.Max = DateTimeOffset.MinValue.AddDays(1);
```
Comment (peppy) [inline osu.Game/Screens/Select/FilterQueryParser.cs:70]: The add one days is just to catch any weirdness? Did you find a specific case? Realm inaccuracy?

Comment (smoogipoo) [reply osu.Game/Screens/Select/FilterQueryParser.cs:70]: `DateTimeOffset.MinValue` is the filtered value:

https://github.com/ppy/osu/blob/a077acc599596fbbfe39ab517961463b34c96310/osu.Game/Screens/Select/Carousel/CarouselBeatmap.cs#L68

An alternative could be...
```
criteria.LastPlayed.Min = DateTimeOffset.MinValue;
criteria.LastPlayed.Max = DateTimeOffset.MinValue;
criteria.LastPlayed.IsLowerInclusive = true;
criteria.LastPlayed.IsUpperInclusive = true;
```
If that makes it better.

Comment (peppy) [reply osu.Game/Screens/Select/FilterQueryParser.cs:70]: Right.

I *think* I prefer that alternative as it's more explicit to what's going on.

**Comment by peppy** [comment]
These implementations are kinda weird aren't they? like you have to type `unplayed=` and providing any argument has no effect?

I wouldn't mind using this as an opportunity to either allow `unplayed` (without the `=`) and/or support `unplayed=false` and `unplayed=true`..

**Comment by peppy** [comment]
Also what... `played=` doesn't do the opposite of `unplayed=`... what a UX.

**Comment by smoogipoo** [comment]
Based on the above two comments I'm not sure what you actually want, so here: https://github.com/ppy/osu/pull/29644

**Pull request closed without merging**